src/components/Homepage/Homepage.js

```diff
@@ -1,8 +1,28 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Link } from 'react-router-dom';
-import 'balloon-css'; 
+import { useSelector, useDispatch } from 'react-redux';
+import { getAllJobs } from '../../helpers/apiCalls'
+import { setJobs, getJobInfo, setErrorMsg, resetErrorMsg } from '../../actions/actions'
+
+import 'balloon-css';
+
+function Homepage({ updateJobAddedStatus, updateLoadingStatus, jobAdded }) {
+  const dispatch = useDispatch();
+  const user = useSelector(state => state.user)
+
+  useEffect(() => {
+      getAllJobs(user.id)
+        .then(data => {
+          dispatch(setJobs(data.data))
+          dispatch(resetErrorMsg());
+          updateJobAddedStatus(false)
+          updateLoadingStatus(true)
+        })
+        .catch(error => {
+          dispatch(setErrorMsg('Sorry, it looks like we are having some trouble retrieving your information. Refresh or try again later.'))
+        })
+  }, [ jobAdded ])
```

Comment:
Definitely makes sense to have Homepage do the fetch. Great refactoring with the props!